app/src/main/java/org/p2p/wallet/auth/ui/restore/WalletFoundPresenter.kt

```diff
@@ -0,0 +1,43 @@
+package org.p2p.wallet.auth.ui.restore
+
+import org.p2p.wallet.auth.common.WalletWeb3AuthManager
+import org.p2p.wallet.auth.common.Web3AuthError
+import org.p2p.wallet.auth.common.Web3AuthHandler
+import org.p2p.wallet.auth.model.GoogleAuthFlow
+import org.p2p.wallet.common.mvp.BasePresenter
+
+class WalletFoundPresenter(
+    private val walletAuthManager: WalletWeb3AuthManager,
+) : BasePresenter<WalletFoundContract.View>(), WalletFoundContract.Presenter {
+
+    private val web3AuthHandler = object : Web3AuthHandler {
+        override fun onSuccessSignUp() {
+            view?.onSuccessfulSignUp()
+        }
+
+        override fun handleError(error: Web3AuthError) {
+            view?.showError(error.errorMessage)
+        }
+    }
+
+    override fun attach(view: WalletFoundContract.View) {
```

Comment:
in some cases view can be null, so it will be more safety, if you use ?, or create method, and call it inside onAttach method

Reply:
you sure? it is overridden method